### DISCUSSION TITLE
Issue with using imagemagick to manipulate pytorch tensors/.tiff files

### DISCUSSION THREAD

**Discussion by AndrewPlewe** [ImageMagick/ImageMagick#6350] [Help]
### ImageMagick version

7-1-1x10

### Operating system, version and so on

Windows

### Description

Hi! I am working with some pytorch tensors that I'm saving, essentially, as .tiff files. The tensor shape is such that it maps to a typical four-channel tiff (these come from the Stable Diffusion UNET diffusion loop). Converting the .tiff to an ftxt gives an output such as:

0,0:-62719,-470777,113214,189693
1,0:-24991.6,-473593,288764,189565
2,0:22463.7,-490488,262652,200829
3,0:20207.7,-487673,260348,195709
4,0:22639.7,-493560,266748,211453
5,0:25167.6,-493560,267004,209789
6,0:19631.7,-494072,256508,201341
7,0:12695.8,-493304,261884,199421
8,0:13743.8,-493048,271356,203389
9,0:11199.8,-497144,268540,203389
10,0:20511.7,-498168,271868,205053
11,0:17311.7,-495864,274940,205565
12,0:6223.91,-494328,278524,200829
13,0:3199.95,-491512,286460,200957

I am running into an issue when manipulating these .tiff files. For instance, just doing a straight-across copy like so:

magick car1_p30_unet.tiff -define quantum:format=floating-point -depth 16 car1_p30_unet_clone.tiff

Somehow messes up the destination .tiff. Both the source and destination are 16 bit per channel per "pixel", and outputting the result via ftxt looks correct:

0,0:-62719,-470777,113214,189693
1,0:-24991.6,-473593,288764,189565
2,0:22463.7,-490488,262652,200829
3,0:20207.7,-487673,260348,195709
4,0:22639.7,-493560,266748,211453
5,0:25167.6,-493560,267004,209789
6,0:19631.7,-494072,256508,201341
7,0:12695.8,-493304,261884,199421
8,0:13743.8,-493048,271356,203389
9,0:11199.8,-497144,268540,203389
10,0:20511.7,-498168,271868,205053
11,0:17311.7,-495864,274940,205565
12,0:6223.91,-494328,278524,200829
13,0:3199.95,-491512,286460,200957

Opening both in Notepad++ and doing a "compare" yields no differences. However, upon importing the .tiff back into pytorch there are differences and the downstream inference process (which is deterministic) using the data is also different. Attached are the two "thumbnails", then the outputs after running an inference step (using the VAE model in Stable Diffusion) on the result. Any idea what I might be doing wrong here? There is a .1 k difference in the file size between the thumbnails, so whatever is different is happening (I assume) in the file headers. Attaching all four in a .zip file as .tiff files aren't directly supported. Thanks!

[latents.zip](https://github.com/ImageMagick/ImageMagick/files/11528628/latents.zip)

**Comment by snibgo** [comment]
I know nothing about pytorch.

As you say, "`magick compare`" says the images are identical. "`exiftool`" and "`magick car1...tiff -verbose info:`" show some differences in metadata. For example, the file created by IM contains sRGB metadata:
```
White Point                     : 0.3127000034 0.3289999962
Primary Chromaticities          : 0.6399999857 0.3300000131 0.3000000119 0.6000000238 0.150000006 0.05999999866
```
I assume this metadata is spurious. Perhaps pytorch uses that metadata, or some other metadata that has changed.

I suggest you try an experiment. Use `magick` to make a copy, then use `exiftool` to replace all the metadata in the copy with metadata from the original. Does that fix the problem? If it does, then we know the problem is in metadata that exiftool can see. A process of elimination should identify which metadata.

**Reply by AndrewPlewe** [reply]
O.k., I will do that. For my purposes that would be enough if it works (the two-step process of running the magick command(s) and then replacing the metadata), although I'm happy to report back what the results are if you think a code change would be useful to others.
